app/Models/Multiplayer/Room.php

```diff
@@ -446,6 +446,19 @@ public function completePlay(ScoreToken $scoreToken, array $params): ScoreLink
                 $stats->save();
             }
 
+            // spotlight playlists should always be linked to one season exactly
+            if ($this->category === 'spotlight' && $agg->total_score > 0 && $this->seasons()->count() === 1) {
+                $seasonId = $this->seasons()->first()->getKey();
+
+                $seasonScore = $user->seasonScores()
+                    ->where('season_id', $seasonId)
+                    ->firstOrNew();
+
+                $seasonScore->season_id = $seasonId;
```

Comment:
`->season()->associate($season)` (this also loads the relation so it's not queried again in calculate)